### ISSUE TITLE
UserAuthenticator: connection timeout undefined !!

### ISSUE THREAD

**Issue by bijaySussol** [openmsupply/mobile#4677]
## Describe the bug

Mobile says: ` LOG  text {"name":"Authentication","hostname":"react-native","level":40,"msg":"UserAuthenticator: connection timeout undefined","time":"2022-04-27T17:49:50.242Z","v":0}` and it does it without anything...

### To reproduce

Steps to reproduce the behaviour:

1. Setup Mobile App
2. Navigate the Customer invoice / Customer requisition
3. See Metro node on prompt
4. You shall see the message

### Expected behaviour

I do not know why Mobile would say the Connection timeout without any intervention, It looks serious for me!!!
It should not, I think !!!!

### Proposed Solution

Leave if you don't know how to fix/implement. Edit this issue description and explain here if you know the best path of implementing the fix within the codebase.

### Version and device info

- App version: 8.4.2
- Tablet model: Nexus 10 Emulator
- OS version: Android 10

### Additional context

See Video:

[Video1](https://www.dropbox.com/s/f5b6wdhey7iet1l/mobile%20crashing%20with%20auto%20logout%20react%20native-1.mp4?dl=1)

[Video2](https://www.dropbox.com/s/f5b6wdhey7iet1l/mobile%20crashing%20with%20auto%20logout%20react%20native-1.mp4?dl=1)

**Comment by bijaySussol** [comment]
Image attached:

<img width="1421" alt="Screen Shot 2022-04-27 at 11 39 12 PM" src="https://user-images.githubusercontent.com/13144882/165591633-8ee12ee4-5760-46b1-a1b9-868b4d64d5a9.png">

**Comment by katherine-sussol** [comment]
🤔  I think this might be a bit of a red-herring kind of log message, haha. I've only played around with it a little bit but it looks like it's coming from here: https://github.com/openmsupply/mobile/blob/19dc0094a4ca231fee8a54f5031f538b82d85a91/src/authentication/UserAuthenticator.js#L134

Which is used here:
https://github.com/openmsupply/mobile/blob/19dc0094a4ca231fee8a54f5031f538b82d85a91/src/authentication/UserAuthenticator.js#L67-L70

What I think is happening is this `createConnectionTimeoutPromise` is being fired off on _every_ authentication. Regardless of whether or not the authentication succeeds the `createConnectionTimeoutPromise` will always be rejected after a fixed period of time, and that message will always be logged.

The intention (I think, just based on my read of it) is that if the connection takes longer than the `createConnectionTimeoutPromise` does to resolve, then the whole thing will just be rejected with a CONNECTION_FAILURE message. 
However if the authentication succeeds before the timeout, then it will resolve with the appropriate authentication response (and the result of the `createConnectionTimeoutPromise` just gets thrown out, but that error message still gets logged). 

Maybe the message itself just needs to be adjusted to be less alarming? ⏰

**Comment by mark-prins** [comment]
yes, you're right @katherine-sussol - the timeout is firing regardless.

a better way could be something like this

```
const createConnectionTimeoutPromise = () => 
   new Promise((resolve, reject) => { 
     const id = setTimeout(() => { 
       reject(new Error('UserAuthenticator: connection timeout')); 
     }, CONNECTION_TIMEOUT_PERIOD); 
   }); 
```

and

```
const handleLogin = (userJson) => userJson;

const handleError = (e) => {
    logger.error(e.message);
   throw(e); // ?? not sure if that's useful, but I think it is
  }

const userJson = await Promise.race([ 
   authenticateAsync(authURL, username, passwordHash, { ...this.extraHeaders }), 
   createConnectionTimeoutPromise(), 
 ]).then(handleLogin, handleError); 
```
